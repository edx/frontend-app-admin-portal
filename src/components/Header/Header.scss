header {
    border-bottom: 1px solid #d4d4d4;

    .navbar-brand {
        max-width: 200px;
    }

    .badge.beta {
        background: #636c72;
        color: #ffffff;
    }
    
    .dropdown > button {
        cursor: pointer;
        color: #636c72;
        background: #ffffff;
    }
}
