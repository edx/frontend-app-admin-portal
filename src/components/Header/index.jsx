import React from 'react';
import PropTypes from 'prop-types';
import { Link } from 'react-router-dom';
import { Dropdown } from '@edx/paragon';

import Img from '../Img';
import EdxLogo from '../../images/edx-logo.png';
import './Header.scss';

const Header = (props) => {
  const {
    enterpriseLogo,
    enterpriseSlug,
    email,
  } = props;

  return (
    <header className="container">
      <nav className="navbar px-0 justify-content-between">
        <Link
          to={enterpriseSlug ? `/${enterpriseSlug}` : '/'}
          className="navbar-brand"
        >
          <Img src={enterpriseLogo || EdxLogo} alt="" />
        </Link>
        {email && <Dropdown
          title={email}
          menuItems={[
            <Link to="/logout">Logout</Link>,
          ]}
        />}
      </nav>
    </header>
  );
};

Header.propTypes = {
  enterpriseSlug: PropTypes.string,
  enterpriseLogo: PropTypes.string,
  email: PropTypes.string,
};

Header.defaultProps = {
  enterpriseSlug: null,
  enterpriseLogo: null,
  email: null,
};

export default Header;
