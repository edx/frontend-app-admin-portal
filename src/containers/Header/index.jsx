import { connect } from 'react-redux';
import Header from '../../components/Header';

const mapStateToProps = state => ({
  enterpriseSlug: state.portalConfiguration.enterpriseSlug,
  enterpriseLogo: state.portalConfiguration.enterpriseLogo,
  email: state.authentication.email,
});

export default connect(mapStateToProps)(Header);
