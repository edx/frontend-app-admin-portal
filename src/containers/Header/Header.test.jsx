import React from 'react';
import PropTypes from 'prop-types';
import renderer from 'react-test-renderer';
import { MemoryRouter } from 'react-router-dom';
import configureMockStore from 'redux-mock-store';
import thunk from 'redux-thunk';

import Header from './index';

const mockStore = configureMockStore([thunk]);

const HeaderWrapper = props => (
  <MemoryRouter>
    <Header
      store={props.store}
      {...props}
    />
  </MemoryRouter>
);

HeaderWrapper.propTypes = {
  store: PropTypes.shape({}).isRequired,
};

describe('<Header />', () => {
  let store;
  let tree;

  it('renders enterprise logo correctly', () => {
    store = mockStore({
      portalConfiguration: {
        enterpriseSlug: 'test-enterprise',
        enterpriseLogo: 'https://test.url/image/1.png',
      },
      authentication: {
        email: 'test@example.com',
      },
    });

    tree = renderer
      .create((
        <HeaderWrapper store={store} />
      ))
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('renders edX logo correctly', () => {
    store = mockStore({
      portalConfiguration: {},
      authentication: {},
    });
    tree = renderer
      .create((
        <HeaderWrapper store={store} />
      ))
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
