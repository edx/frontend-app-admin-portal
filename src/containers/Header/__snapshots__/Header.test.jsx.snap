// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Header /> renders edX logo correctly 1`] = `
<header
  className="container"
>
  <nav
    className="navbar px-0 justify-content-between"
  >
    <a
      className="navbar-brand"
      href="/"
      onClick={[Function]}
    >
      <img
        alt=""
        className=""
        src="test-file-stub"
      />
    </a>
  </nav>
</header>
`;

exports[`<Header /> renders enterprise logo correctly 1`] = `
<header
  className="container"
>
  <nav
    className="navbar px-0 justify-content-between"
  >
    <a
      className="navbar-brand"
      href="/test-enterprise"
      onClick={[Function]}
    >
      <img
        alt=""
        className=""
        src="https://test.url/image/1.png"
      />
    </a>
    <div
      className="dropdown"
    >
      <button
        aria-expanded={false}
        aria-haspopup="true"
        className="btn dropdown-toggle btn-light"
        onBlur={[Function]}
        onClick={[Function]}
        onKeyDown={[Function]}
        type="button"
      >
        test@example.com
      </button>
      <div
        aria-hidden={true}
        aria-label="test@example.com"
        className="dropdown-menu"
        role="menu"
      >
        <a
          className="dropdown-item"
          href="/logout"
          onClick={[Function]}
          onKeyDown={[Function]}
        >
          Logout
        </a>
      </div>
    </div>
  </nav>
</header>
`;
